### PR TITLE
Tubs18 team2

### PR DESCRIPTION
**Pull request by BassMonster**
Solves the issues https://github.com/FeatureIDE/FeatureIDE/issues/763 and https://github.com/FeatureIDE/FeatureIDE/issues/765 and now you can select a whole subtree

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/FeatureDiagramEditorKeyHandler.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public FeatureDiagramEditorKeyHandler(ScrollingGraphicalViewer view, IGraphicalF
 		viewer = view;
 
 		alternativeKeyHandler = new KeyHandler();
-		gvKeyHandler = new GraphicalViewerKeyHandler(view);
+		gvKeyHandler = new MovingKeyHandler(view);// GraphicalViewerKeyHandler(view);
```
Comment (KamilRosiak) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/FeatureDiagramEditorKeyHandler.java]: Remove code comment

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/MovingKeyHandler.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		final EditPart part = getViewer().getFocusEditPart();
+		final Map<?, ?> editPartRegistry = getViewer().getEditPartRegistry();
+
+		// true if a feature is selected
+		if (part instanceof FeatureEditPart) {
+			final FeatureEditPart featurepart = (FeatureEditPart) part;
+			final IGraphicalFeature feature = featurepart.getModel();
+			final IGraphicalFeature parent = feature.getSourceConnection().getTarget();
+			final int dex = parent != null ? parent.getObject().getStructure().getChildIndex(feature.getObject().getStructure()) : 0;
+			if (!feature.getGraphicalModel().getLayout().verticalLayout()) {
```
Comment (KamilRosiak) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/MovingKeyHandler.java:70]: Add some comments for the decisions.

**Review by KamilRosiak — COMMENTED** [review]
Code looks good just add some comments for the decision.
@PaulWestphal

**Comment by BassMonster** [comment]
@PaulWestphal we applied the requested changes

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/MovingKeyHandler.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+/**
+ * TODO description
+ *
+ * @author Sabrina Hugo
+ * @author Christian Orsinger
+ */
+public class MovingKeyHandler extends GraphicalViewerKeyHandler {
+
+	public enum direction {
+		left, right
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/keyhandler/MovingKeyHandler.java:46]: Should be upper case.

**Review by PaulWestphal — APPROVED** [review]

**Pull request merged**